core/import-seac.c

```diff
@@ -303,3 +308,15 @@ int parse_seac_buffer(sqlite3 *handle, const char *url, const char *buffer, int
 
 	return 0;
 }
+
+
+// The Practice of Programming (HASH TABLES, pg. 57)
```

Comment:
that doesn't follow our coding standard at all.
I understand the desire to cut and paste from a source - but if you do that, you also have to follow the LICENSE of that code - and those are typically not GPL compatible when it comes to books...
We already have a `calculate_string_hash()` - or is there a reason why this needs to be a specific hash function?

Reply:
No particular reason, I wasn't aware of the existing hash function.

I've already swapped it over to the hash function in `divecomputer.h`
You'll have to remind me, is the preference to rebase then force push so that the pull appears as a single commit?

Reply:
That's an oversimplification :)
The preference is to have a "sane" git history.
We are happy to have multiple commits in a PR, but we prefer that they don't introduce something and then rip it back out (unless there is a REASON for that... e.g., you move from one infrastructure to a different one, you put in some helpers to make it possible to cleanly do this in multiple steps, and then remove the helpers).

Here, given that this really does seem to be one self-contained single step, I'd rebase and force push, TBH

Reply:
Thanks for the information; that makes a lot of sense.

I have force pushed the version using `calculate_string_hash()`

Reply:
Just one additional remark: Please make sure that with every commit the build succeeds and it is possible to run the program as otherwise a future git bisect can be painful.